index.d.ts

```diff
@@ -0,0 +1,26 @@
+import {Url} from "url";
+import {IncomingMessage} from "http";
+
+export interface HttpieOptions {
+	body: any,
+	headers: {
+		[name: string]: string
+	},
+	redirect: boolean
+}
+
+export interface HttpieResponse<T> extends IncomingMessage {
```

Comment:
I'm not super knowledgeable with TS, but I _think_ that rather than: `data: string | T` we could accomplish this with `data: T` if we do: `HttpieResponse<T = string>`. This would have the added benefit of treating the string as a type error if something else is specified.

Reply:
Hmm.. are you saying that it will throw an error if `data` ends up being an object or array? When working with JSON endpoints, it will almost always be either of these since `httpie` parses automatically.

Reply:
Well, with the way these types are written, it looks like the `data` property can be either a string or whatever generic type people provide this interface. With what you've said it should probably be:

```typescript
export interface HttpieResponse<T = any> extends IncomingMessage {
	data: T,
}
```

But again, I'm far from a TS guru.

Reply:
Ah, that makes sense. Thank you – you've sure been a TS guru for me 😉

Reply:
it was `T | string` because it could unexpectedly be a string